programs/mango-v4/src/state/bank.rs

```diff
@@ -693,16 +733,40 @@ impl Bank {
         allow_dusting: bool,
         now_ts: u64,
     ) -> Result<WithdrawResult> {
-        let opening_indexed_position = position.indexed_position;
-        let res = self.withdraw_internal(
-            position,
-            native_amount,
-            with_loan_origination_fee,
-            allow_dusting,
-            now_ts,
-        );
-        self.update_cumulative_interest(position, opening_indexed_position);
-        res
+        if position.allow_lending() {
+            assert!(position.unlendable_deposits == 0);
+            let opening_indexed_position = position.indexed_position;
+            let res = self.withdraw_internal(
```

Comment:
Same here, would rename `withdraw_internal` to a more explicit `withdraw_indexed_pos_internal` or `withdraw_lendable_pos_internal`